File: src/ru/edu/vtisov/array/twopointers/ContainerWithMostWater.java
package ru.edu.vtisov.array.twopointers;

// https://leetcode.com/problems/container-with-most-water

public class ContainerWithMostWater {
    public int maxArea(int[] height) {
        int max = 0;
        for (int i = 0, j = height.length - 1; i < j;) {
            max = Math.max(Math.min(height[i], height[j]) * (j - i), max);
            if (height[i] < height[j]) i++;
            else j--;
        }
        return max;
    }
}

File: src/ru/edu/vtisov/array/twopointers/container-with-most-water.go
package twopointers

func maxArea(height []int) int {
	result := 0
	i := 0
	j := len(height) - 1
	for i < j {
		result = max(result, (j-i)*min(height[i], height[j]))
		if height[i] < height[j] {
			i++
		} else {
			j--
		}
	}
	return result
}
